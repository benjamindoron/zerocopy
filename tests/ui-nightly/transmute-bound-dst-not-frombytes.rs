// Copyright 2022 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.
extern crate zerocopy;

include!("../../zerocopy-derive/tests/util.rs");

// The source type must be `FromBytes`.
const DST_NOT_FROM_BYTES: NotZerocopy<AU16> = zerocopy::transmute!(AU16(0));

fn main() {
    _ = DST_NOT_FROM_BYTES;
}
