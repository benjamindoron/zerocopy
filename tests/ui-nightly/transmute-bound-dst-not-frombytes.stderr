error[E0277]: the trait bound `NotZerocopy<AU16>: FromBytes` is not satisfied
 --> tests/ui-nightly/transmute-bound-dst-not-frombytes.rs:9:47
  |
9 | const DST_NOT_FROM_BYTES: NotZerocopy<AU16> = zerocopy::transmute!(AU16(0));
  |                                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `FromBytes` is not implemented for `NotZerocopy<AU16>`
  |
  = help: the following other types implement trait `FromBytes`:
            ()
            AU16
            F32<O>
            F64<O>
            I128<O>
            I16<O>
            I32<O>
            I64<O>
          and $N others
note: required by a bound in `DST_NOT_FROM_BYTES::transmute`
 --> tests/ui-nightly/transmute-bound-dst-not-frombytes.rs:9:47
  |
9 | const DST_NOT_FROM_BYTES: NotZerocopy<AU16> = zerocopy::transmute!(AU16(0));
  |                                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `transmute`
  = note: this error originates in the macro `zerocopy::transmute` (in Nightly builds, run with -Z macro-backtrace for more info)
